datafusion/core/tests/user_defined/array_has.rs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::datatypes::UInt32Type;
+use arrow::datatypes::{DataType, Field, Schema};
+use arrow::record_batch::RecordBatch;
+use arrow_array::ListArray;
+use datafusion::logical_expr::ScalarUDF;
+use datafusion::prelude::*;
+use datafusion_functions_nested::array_has::ArrayHas;
+use std::sync::Arc;
+
+/// can't go next to the `ArrayHas` struct due to circular dependencies, hence this standalone test
```

Comment:
Since this test isn't a user defined function per se, can we move this test to https://github.com/apache/datafusion/tree/main/datafusion/core/tests/sql instead ?

Reply:
test moved.